src/compas_timber/connections/solver.py

```diff
@@ -372,3 +371,55 @@ def does_segment_intersect_outline(segment, polyline, tol=TOL):
         if intersection_segment_polyline(segment, polyline, tol.absolute)[0]:
             return True
         return is_point_in_polyline(segment.point_at(0.5), polyline, in_plane=False, tol=tol)
+
+class BeamConnectionTopologyResults(Data):
+    """Data structure to hold the results of beam connection topology analysis."""
+
+    def __init__(self, topology, beam_a, beam_b, distance=None, location=None):
+        """Initializes the BeamConnectionTopologyResults with the given parameters.
+
+        Parameters
+        ----------
+        topology : :class:`~compas_timber.connections.JointTopology`
+            The topology of the intersection.
+        beam_a : :class:`~compas_timber.parts.Beam`
+            The first beam involved in the intersection.
+        beam_b : :class:`~compas_timber.parts.Beam`
+            The second beam involved in the intersection.
+        distance : float
+            The distance between the closest points of the two beams.
+        location : :class:`~compas.geometry.Point`
+            The location of the intersection.
+        """
```

Comment:
`__init__` doesn't have its own docstring. instead, the docstring should be on the class